src/components/Common/Alert/Alert.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useId, useRef } from 'react'
+import styles from './Alert.module.scss'
+import InfoIcon from '@/assets/icons/info.svg?react'
+import SuccessIcon from '@/assets/icons/success_check.svg?react'
+import WarningIcon from '@/assets/icons/warning.svg?react'
+import ErrorIcon from '@/assets/icons/error.svg?react'
+
+export interface AlertProps {
+  variant?: 'info' | 'success' | 'warning' | 'error'
+  label: string
+  children?: React.ReactNode
+  icon?: React.ComponentType<React.SVGProps<SVGSVGElement> & { title?: string }>
+}
+
+export function Alert({ label, children, variant = 'info', icon }: AlertProps) {
+  const id = useId()
+  const alertRef = useRef<HTMLDivElement>(null)
+  const IconComponent = icon
+    ? icon
+    : variant === 'info'
+      ? InfoIcon
+      : variant === 'success'
+        ? SuccessIcon
+        : variant === 'warning'
+          ? WarningIcon
+          : ErrorIcon
+
+  useEffect(() => {
+    if (alertRef.current) alertRef.current.scrollIntoView({ behavior: 'smooth' })
+  }, [])
```

Comment:
I was initially surprised to see no effect dependency on `alertRef.current` here although that may be synaptic misfire on my part :)